metrics/web/docs/03_widget_structure_organization.md

```diff
@@ -1,54 +1,350 @@
+
+![Metrics Theme Structure Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/software-platform/monorepo/widget_stucture_organization_document/metrics/web/docs/diagrams/theme_data_class_diagram.puml)
+
+The main idea of the Metrics Theme inspired by Flutter default MaterialTheme that provides the `InheritedWidget` with the theme data to widgets. So, we have a `MetricsThemeData` class that contains the theme data for all the Metric widgets. Also, we have a `MetricsTheme` widget - the `InheritedWidget` that provides the `MetricsThemeData` to descendant widgets. To simplify the mechanism of changing the theme (from light to dart), there is the `MetricsThemeBuilder` widget that holds the light and dark themes and builds the MetricsTheme widget depending on the current theme state.
+
+See the diagram below for a more detailed description of metrics theme structure: 
+
+![Metrics Theme Structure Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/software-platform/monorepo/widget_stucture_organization_document/metrics/web/docs/diagrams/metrics_theme_structure_diagram.puml)
+
+### Applying a Theme to a widget appearance [Pending]
 > Explain the algorithm of applying the metrics theme to the widgets.
 
+The main concept of applying the themes in the Metrics Web Application is to create a separate theme data or a separate field in `MetricsThemeData` class with `MetricWidgetThemeData` type for each high-level widget. The low-level widgets should not apply any theme. If the low-level widget requires any default colors, we should set them in the constructor default params or create some constants, but not use the metrics theme to make the low-level widgets independent of the Metrics Web Application context.
+
+So, the low-level widget should have the color params in the constructor, and the high-level widget that uses this low-level widget should apply the appropriate theme for it. This means that the appearance of the low-level widget is always controlled outside of it.
+
+If widgets require the custom theme (different from `MetricWidgetThemeData`, or any existing ones), we should create a new theme data (see [Adding a new Theme](#Adding_a_new_Theme)), specific for this widget. All the theme data classes should be stored in a `common/presentation/metrics_theme/model` folder. Let's consider the activity diagram that will explain the process of applying a theme data to a widget: 
+
+![Apply Widget Theme Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/software-platform/monorepo/widget_stucture_organization_document/metrics/web/docs/diagrams/apply_widget_theme_diagram.puml)
```

Comment:
Not sure about the path for low-level widgets:
![image](https://user-images.githubusercontent.com/442086/83278311-124af200-a1dc-11ea-9ccc-e4740b55343b.png)
Should we just end the process if it's low-level?

Reply:
Do we still need this step about active/inactive on the diagram?